pallets/creator-staking/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+    pub type NextEraStartingBlock<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn registered_creator)]
+    pub(crate) type RegisteredCreators<T: Config> =
+        StorageMap<_, Twox64Concat, SpaceId, CreatorInfo<T::AccountId>>;
+
+    /// Staking information about creator in a particular era.
+    #[pallet::storage]
+    #[pallet::getter(fn creator_stake_info)]
+    pub type CreatorEraStake<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        SpaceId,
+        Twox64Concat,
+        EraIndex,
+        CreatorStakeInfo<BalanceOf<T>>,
+    >;
+
+    /// Info about stakers stakes on particular creators.
+    #[pallet::storage]
+    #[pallet::getter(fn staker_info)]
+    pub type GeneralStakerInfo<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        T::AccountId,
+        Blake2_128Concat,
+        SpaceId,
+        StakerInfo<BalanceOf<T>>,
+        ValueQuery,
+    >;
+
+    /// General information about an era like TVL, total staked value, rewards.
+    #[pallet::storage]
+    #[pallet::getter(fn general_era_info)]
+    pub type GeneralEraInfo<T: Config> =
+        StorageMap<_, Twox64Concat, EraIndex, EraInfo<BalanceOf<T>>>;
+
+    /// General information about the staker.
+    #[pallet::storage]
+    #[pallet::getter(fn ledger)]
+    pub type Ledger<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AccountId, AccountLedger<BalanceOf<T>>, ValueQuery>;
+
+    /// Accumulator for block rewards during an era. It is reset at every new era
+    #[pallet::storage]
+    #[pallet::getter(fn block_reward_accumulator)]
+    pub type BlockRewardAccumulator<T> = StorageValue<_, RewardInfo<BalanceOf<T>>, ValueQuery>;
+
+    #[pallet::type_value]
+    pub fn RewardConfigOnEmpty() -> RewardDistributionConfig {
+        RewardDistributionConfig::default()
+    }
+
+    /// An active list of configuration parameters used to calculate reward distribution portions.
+    #[pallet::storage]
+    #[pallet::getter(fn reward_config)]
+    pub type ActiveRewardDistributionConfig<T> =
+        StorageValue<_, RewardDistributionConfig, ValueQuery, RewardConfigOnEmpty>;
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub (super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        Staked { who: T::AccountId, creator: SpaceId, era: EraIndex, amount: BalanceOf<T> },
+        Unstaked { who: T::AccountId, creator: SpaceId, era: EraIndex, amount: BalanceOf<T> },
+        RewardsClaimed { who: T::AccountId, amount: BalanceOf<T> },
+        WithdrawnFromClaimed { who: T::AccountId, amount: BalanceOf<T> },
+        WithdrawnFromUnregistered { who: T::AccountId, amount: BalanceOf<T> },
+        AnnualInflationSet { value: Perbill },
+        RewardsCalculated { total_rewards_amount: BalanceOf<T> },
+        CreatorRegistered { who: T::AccountId, space_id: SpaceId },
+        CreatorUnregistered { who: T::AccountId, space_id: SpaceId },
```

Comment:
Let's return `amount` here as well, becasue this event is pretty similar to event `CreatorUnregisteredWithSlash`.
```suggestion
        CreatorUnregistered { who: T::AccountId, space_id: SpaceId, amount: BalanceOf<T> },
```

Reply:
We pass `amount` to `...WithSlash`, because we want to show the slash itself. Nothing to show here unless we want to show the deposit (which isn't dynamic and will always be a constant `CreatorRegistrationDeposit`). Thus redundant.